lib/src/merge.rs

```diff
@@ -249,25 +249,83 @@ impl<T> Merge<T> {
         }
     }
 
-    /// Simplify the merge by joining diffs like A->B and B->C into A->C.
-    /// Also drops trivial diffs like A->A.
-    pub fn simplify(mut self) -> Self
+    /// Returns a vector mapping of a value's index to an [`Option`] of
+    /// its index in the simplified merge. If a given index in the map has the
+    /// value [`None`], the corresponding merge value was removed as part of
+    /// simplification.
+    ///
+    /// The merge is simplified by removing identical values in add and remove
+    /// values.
```

Comment:
Would it be simpler to return a mapping from the simplified index to the unsimplified index? I think that's what `original_indices` is below, so I think I'm asking if we could just return that instead.

So for `A-B+C-A+D` instead of `[None, Some(1), Some(2), None, Some(0)]` we would get `[4,1,2]`.

Reply:
I think it would be if we could assign a new vector to `self.values` instead of modifying it in-place. I wasn't sure if that was desired.

Reply:
Actually, won't it be safe to use `slice::swap()` there because elements are never moved later in the list? If I'm wrong about that, it seems fine to copy the values over to a new vector.

Reply:
Elements can be moved later in the list, so I think just using `slice::swap()` doesn't work. I've updated the PR to clone the values to the new vector for now.